iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -823,7 +823,7 @@ public override async Task EnableMethodsAsync(CancellationToken cancellationToke
             // Codes_SRS_CSHARP_MQTT_TRANSPORT_18_001:  `EnableMethodsAsync` shall subscribe using the '$iothub/methods/POST/' topic filter.
             // Codes_SRS_CSHARP_MQTT_TRANSPORT_18_002:  `EnableMethodsAsync` shall wait for a SUBACK for the subscription request.
             // Codes_SRS_CSHARP_MQTT_TRANSPORT_18_003:  `EnableMethodsAsync` shall return failure if the subscription request fails.
-            await _channel.WriteAsync(new SubscribePacket(0, new SubscriptionRequest(MethodPostTopicFilter, QualityOfService.AtMostOnce))).ConfigureAwait(true);
+            await _channel.WriteAsync(new SubscribePacket(0, new SubscriptionRequest(MethodPostTopicFilter, QualityOfService.AtLeastOnce))).ConfigureAwait(true);
```

Comment:
Should we never use `QualityOfService.AtMostOnce` in our SDK? Should we put a comment on that option to discourage it (or is that type from the MQTT library)?

Reply:
I was thinking along the same lines that we should always be avoiding QoS of 0, but turns out we do still publish messages with QoS of 0. I'm digging into it deeper to see if this should be changed as well (us publishing messages with QoS 1 will require that service ack those messages).